internal/bminventory/inventory_test.go

```diff
@@ -9177,9 +9192,98 @@ var _ = Describe("V2DownloadInfraEnvFiles", func() {
 			os.Unsetenv("EC_PRIVATE_KEY_PEM")
 		})
 
+		Context("Skipped IPXE", func() {
+			scenarios := []struct {
```

Comment:
I very much don't like this style of test.
I think it ruins one of the best parts of BDD frameworks which is how the test clearly describes the behavior that is expected.

Also this becomes very difficult to maintain if only one of these cases needs some kind of different handling in the future.

I think I'd prefer to have a few different examples like this intead of this scenario struct and a loop:
```go
It("doesn't serve the ipxe script when the host is installing or installed", func() {
  // Test all the InstallingInProgress/Installed cases here
})

It("serves the ipxe script when the host is not installing", func() {
  // Test the other cases here
})
```

Reply:
I previously received comments that I am not running similar tests in loop.  We need to have single approach for such tests